src/test/java/pages/RestaurantsPage.java

```diff
@@ -19,43 +19,63 @@ public class RestaurantsPage {
 
     public final HeaderGeneralPageComponent headerGlobal;
 
-    public RestaurantsPage(WebDriver driver)    {
+    public RestaurantsPage(WebDriver driver) {
         PageFactory.initElements(driver, this);
         this.driver = driver;
         headerGlobal = new HeaderGeneralPageComponent(driver);
     }
+
     @FindBy(xpath = "//main/div/div/div")
     private List<WebElement> restaurants;
 
-    private WebElement neededRestaurant(String restName)   {
-        wait = new WebDriverWait(driver, Duration.ofSeconds(3));
+    private WebElement neededRestaurant(String restName) {
+        WebElement neededRestaurant = restaurants.get(0);
+        for (WebElement restaurant : restaurants) {
+            if (restaurant.findElement(By.xpath(
+                            ".//div/div"))
+                    .getAttribute("title").equals(restName)) {
+                neededRestaurant = restaurant;
+            }
+        }
+        return neededRestaurant;
+    }
+
+    private WebElement neededRestaurant(String restName, Duration duration) {
+        wait = new WebDriverWait(driver, duration);
         // Using for loop, it tries for 3 times.
         // If the element is located for the first time then it breaks from the for loop nad comeout of the loop
-        for(int i=0; i<=2;i++){
-            try{
+        for (int i = 0; i <= 2; i++) {
+            try {
                 wait.until(ExpectedConditions.refreshed(ExpectedConditions.stalenessOf(restaurants.get(0))));
                 break;
-            }
-            catch(Exception e){
+            } catch (Exception e) {
 
             }
         }
-
-
         WebElement neededRestaurant = restaurants.get(0);
-        for(WebElement restaurant : restaurants)    {
-            if(restaurant.findElement(By.xpath(
+        for (WebElement restaurant : restaurants) {
+            if (restaurant.findElement(By.xpath(
                             ".//div/div"))
-                    .getAttribute("title").equals(restName))   {
+                    .getAttribute("title").equals(restName)) {
                 neededRestaurant = restaurant;
             }
         }
         return neededRestaurant;
     }
 
-    public MenuPage watchMenuByRestName(String restName)    {
+    public MenuPage watchMenuByRestName(String restName) {
         neededRestaurant(restName).findElement(By.xpath(
+                    ".//*[text()='Watch Menu']")).click();
+        return new MenuPage(driver);
+        }
+
+    public MenuPage watchMenuByRestName(String restName, Duration duration) {
+        neededRestaurant(restName, duration).findElement(By.xpath(
                 ".//*[text()='Watch Menu']")).click();
         return new MenuPage(driver);
     }
-}
+
+    }
```

Comment:
looks like you have a wrong number of closed brackets "}" or it's just the wrong position, please review that

Reply:
It`s just the wrong position, I will fix it.